### ISSUE TITLE
[CSP Adapter] Support for Custom Certs

### ISSUE THREAD

**Issue by MbolotSuse** [rancher/rancher#38050]
The CSP adapter should support communicating with rancher using the current supported methods for specifying a cert, namely:

- Rancher Generated
- Let's encrypt
- Specified Secret

Regardless of the method chosen, the adapter should be configured to trust the cert used by rancher, and be able to securely communicate with rancher over tls.

**Comment by MbolotSuse** [comment]
This should be available in 1.0.0-rc2 of the adapter.

**Comment by brudnak** [comment]
# :white_check_mark: PASSED

## Reproduction Environment

#### Not required.

<hr>
  
## Validation Environment

| Component               | Version / Type                                                                     |
| ----------------------- | ---------------------------------------------------------------------------------- |
| Rancher version         | v2.6-bda0897a42ac77431383eb4a4bcd0454de056b98-head                                 |
| Rancher commit link     | https://github.com/rancher/rancher/commit/bda0897a42ac77431383eb4a4bcd0454de056b98 |
| Installation option     | `eksctl` eks local cluster (bring your own cluster)                                |
| local k8s cluster       | v1.22.10-eks-84b4fe6                                                               |
| Cert Details            | let's encrypt                                                                      |
| Helm version            | v2.16.8-rancher2                                                                   |
| Downstream cluster type | rke1 linode                                                                        |
| Downstream K8s version  | v1.23.6-rancher1-1                                                                 |
| Logged in user role     | administrator                                                                      |
| Browser type            | google chrome                                                                      |
| Browser version         | 102.0.5005.115 (Official Build) (x86_64)                                           |
| CSP Adapter version     | rancher-csp-adapter:1.0.0+up1.0.0-rc2                                              |

**ADDITIONAL SETUP**

- eksctl installation commands used

```shell
eksctl create cluster \
  --name $CLUSTER_NAME \
  --version 1.22 \
  --region $REGION_VAR \
  --nodegroup-name $NODE_GROUP_NAME \
  --nodes 3 \
  --nodes-min 1 \
  --nodes-max 4 \
  --managed
```

- Helm commands for Rancher installation: (let's encrypt example)

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$HOSTNAME_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=nginx \
  --set extraEnv\[0\].name="CATTLE_PROMETHEUS_METRICS" --set-string extraEnv\[0\].value=true
```

## Validation steps

1. Create an EKS Rancher server `(BYOC) bring your own cluster`
2. Setup OIDC & IAM roles as outlined in the documentation
3. Install Rancher with various certificate options Using all three options listed under `step #5`:`Install Rancher with Helm and Your Chosen Certificate Option` https://rancher.com/docs/rancher/v2.6/en/installation/install-rancher-on-k8s/#5-install-rancher-with-helm-and-your-chosen-certificate-option

<table>
    <tbody>
        <tr>
            <th>Cert</th>
            <th>Pass/Fail</th>
        </tr>
        <tr>
            <td>rancher-generated</td>
            <td>:white_check_mark:</td>
        </tr>
        <tr>
            <td>let's encrypt</td>
            <td>:white_check_mark:</td>
        </tr>
        <tr>
            <td>certificate from files</td>
            <td>:white_check_mark:</td>
        </tr>
    </tbody>
</table>

4. Install the CSP Adapter
5. Verify adapter is working as expected, goes into and out of compliance. Generates expected support configs
  
## Additional Info

## RESULTS

#### :white_check_mark: Expected
Expected the csp adapter to work with various cert options

#### :white_check_mark:Actual
The csp adapter works with various cert options
